Applications/Custom/mae_loss.h

```diff
@@ -14,82 +14,80 @@
 #define __MAE_LOSS_LAYER_H__
 #include <string>
 
-/// @todo migrate these to API and ensure those headers are exposed to devel
-#include <layer_internal.h>
-#include <manager.h>
-#include <tensor.h>
+#include <layer_context.h>
+#include <layer_devel.h>
+#include <node_exporter.h>
 
 namespace custom {
 
 /**
  * @brief A sample loss layer which calculates mean absolute error from output
- * @todo update this to LayerV1
  *
  */
-class MaeLossLayer final : public nntrainer::LayerV1 {
+class MaeLossLayer final : public nntrainer::Layer {
 public:
   /**
-   * @brief Construct a new Pow Layer object that does elementwise power
+   * @brief Construct a new MAE Layer object that does elementwise power
    *
    */
-  MaeLossLayer() : LayerV1() {}
+  MaeLossLayer() : Layer() {}
 
   /**
-   * @brief Destroy the Pow Layer object
+   * @brief Destroy the MAE Layer object
    *
    */
   ~MaeLossLayer() {}
 
-  using nntrainer::LayerV1::setProperty;
-
   /**
-   * @brief     set Property of layer, currently only "exponent is accepted"
-   * @param[in] values values of property
-   * @retval #ML_ERROR_NONE Successful.
-   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   * @copydoc Layer::finalize(InitLayerContext &context)
    */
-  int setProperty(std::vector<std::string> values) override {
-    /// this implementation makes to pass the test, this will change soon.
-    return values.size();
+  void finalize(nntrainer::InitLayerContext &context) override {
+    context.setOutputDimensions(context.getInputDimensions());
+    /** NYI */
   }
 
   /**
-   * @brief initializing nntrainer
-   *
-   * @return int ML_ERROR_NONE if success
+   * @copydoc Layer::forwarding(RunLayerContext &context, bool training)
    */
-  int initialize(nntrainer::Manager &manager) override { /** NYI */
-    return 0;
-  }
+  void forwarding(nntrainer::RunLayerContext &context, bool training) override{
+    /** NYI */
```

Comment:
FYI) Afaik, this was already implemented in main.

Reply:
I will copy from main then, rather than relying on merging.